coordinator/src/services/translation/spine-response.ts

```diff
@@ -1,22 +1,51 @@
 import * as fhir from "../../models/fhir/fhir-resources"
 import {CodeableConcept} from "../../models/fhir/fhir-resources"
 import {readXml} from "../serialisation/xml"
-import {acknowledgementCodes, AsyncMCCI, SyncMCCI} from "../../models/hl7-v3/hl7-v3-spine-response"
-import {toArray} from "./common"
+import {
+  acknowledgementCodes,
+  AsyncMCCI, PORX50101,
+  SyncMCCI
+} from "../../models/hl7-v3/hl7-v3-spine-response"
 import {SpineDirectResponse} from "../../models/spine"
+import {translateSpineCancelResponseIntoBundle} from "./cancellation/cancellation-response"
+import {toArray} from "./common"
 
 const SYNC_SPINE_RESPONSE_MCCI_REGEX = /(?=<MCCI_IN010000UK13>)([\s\S]*)(?<=<\/MCCI_IN010000UK13>)/i
 const ASYNC_SPINE_RESPONSE_MCCI_REGEX = /(?=<hl7:MCCI_IN010000UK13[\s\S]*>)([\s\S]*)(?<=<\/hl7:MCCI_IN010000UK13>)/i
+// eslint-disable-next-line max-len
+export const SPINE_CANCELLATION_ERROR_RESPONSE_REGEX = /(?=<hl7:PORX_IN050101UK31[\s\S]*>)([\s\S]*)(?<=<\/hl7:PORX_IN050101UK31>)/i
 
 interface TranslatedSpineResponse {
-  operationOutcome: fhir.OperationOutcome
+  body: fhir.OperationOutcome | fhir.Bundle
   statusCode: number
 }
 
-export function translateToOperationOutcome<T>(message: SpineDirectResponse<T>): TranslatedSpineResponse {
+function isCancellationErrorResponse(message:string) {
+  const cancelResponse = SPINE_CANCELLATION_ERROR_RESPONSE_REGEX.exec(message)
+  if (cancelResponse) {
+    const parsedMsg = readXml(cancelResponse[0]) as PORX50101
+    const parsedMsgAcknowledgement = parsedMsg["hl7:PORX_IN050101UK31"]["hl7:acknowledgement"]
+    const ackCode = parsedMsgAcknowledgement._attributes.typeCode
+    const statusCode = translateAcknowledgementTypeCodeToStatusCode(ackCode)
+    if(statusCode == 400) {
```

Comment:
Do successful cancellations not get returned as a bundle?

Reply:
done